actions/argo-lint/action.yaml

```diff
@@ -0,0 +1,75 @@
+name: Lint Argo Workflow files
+description: Lint Argo workflow files
+
+inputs:
+  artifact_name:
+    description: |
+      Name of the artifact containing the files to lint.
+    required: true
+  path:
+    description: |
+      Path to files within the artifact to lint.
+    default: ""
+
+runs:
+  using: composite
+
+  steps:
+    - name: Set argo version
+      id: argo-version
+      shell: sh
+      run: |
+        ARGO_VERSION=3.5.1
+        echo "version=${ARGO_VERSION}" >> ${GITHUB_OUTPUT}
+
+    - name: Restore cache
+      id: restore
+      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+      with:
+        path: ${{ github.workspace }}/bin
+        key: argo-linux-amd64-${{ steps.argo-version.outputs.version }}
+
+    - name: Fetch Github Release Asset
+      id: fetch_asset
+      if: steps.restore.outputs.cache-hit != 'true'
+      uses: dsaltares/fetch-gh-release-asset@a40c8b4a0471f9ab81bdf73a010f74cc51476ad4 # 1.1.1
+      with:
+        repo: "argoproj/argo-workflows"
+        version: "tags/v${{ steps.argo-version.outputs.version }}"
+        file: "argo-linux-amd64.gz"
+        target: ${{ github.workspace }}/bin/argo.gz
+
+    - name: gunzip
+      id: gunzip
+      if: steps.fetch_asset.outcome == 'success'
+      shell: sh
+      run: |
+        gunzip ${{ github.workspace }}/bin/argo.gz
+        chmod +x ${{ github.workspace }}/bin/argo
+
+    - name: Save to cache
+      id: save
+      if: steps.gunzip.outcome == 'success'
+      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+      with:
+        path: ${{ github.workspace }}/bin
+        key: ${{ steps.restore.outputs.cache-primary-key }}
+
+    - name: Add binary to path
+      shell: sh
+      run: |
+        echo "${{ github.workspace }}/bin" >> $GITHUB_PATH
+
+    - name: Download artifact
+      uses: actions/download-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+      with:
+        name: ${{ inputs.artifact_name }}
+        path: _shared-workflows-argo-lint
```

Comment:
If this is the same as `trigger-argo-workflow`, could we factor it out? Or merge the two actions somehow? (keeping the existing interface)

Reply:
Done, separate action workflow now.

Also changed the argo-lint action to use a path.